Saving the linked test here as there is a TTL to it and future reference to it may not be possible after TTL expires.
[48.txt](https://github.com/user-attachments/files/30463237/48.txt)


Exact error:

```log
2026-07-28T07:43:50.0726429Z [INFO] Running org.apache.hudi.hive.util.TestHiveDriverPool
2026-07-28T07:43:50.0790769Z [ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.005 s <<< FAILURE! -- in org.apache.hudi.hive.util.TestHiveDriverPool
2026-07-28T07:43:50.0791355Z [ERROR] org.apache.hudi.hive.util.TestHiveDriverPool.awaitAllThrowsFirstError -- Time elapsed: 0.005 s <<< FAILURE!
2026-07-28T07:43:50.0791904Z org.opentest4j.AssertionFailedError: Expected org.apache.hudi.hive.HoodieHiveSyncException to be thrown, but nothing was thrown.
2026-07-28T07:43:50.0792734Z 	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:152)
2026-07-28T07:43:50.0793265Z 	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:73)
2026-07-28T07:43:50.0793793Z 	at org.junit.jupiter.api.AssertThrows.assertThrows(AssertThrows.java:35)
2026-07-28T07:43:50.0794314Z 	at org.junit.jupiter.api.Assertions.assertThrows(Assertions.java:3128)
2026-07-28T07:43:50.0795030Z 	at org.apache.hudi.hive.util.TestHiveDriverPool.awaitAllThrowsFirstError(TestHiveDriverPool.java:139)
2026-07-28T07:43:50.0795568Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2026-07-28T07:43:50.0796034Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
2026-07-28T07:43:50.0796541Z 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
```

--- a/hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/HiveDriverPool.java
+++ b/hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/HiveDriverPool.java
@@ -265,7 +265,7 @@ public class HiveDriverPool implements AutoCloseable {
     private final int total;
     private final AtomicInteger settled = new AtomicInteger(0);
     private final AtomicBoolean aborted = new AtomicBoolean(false);
-    private final AtomicReference<Throwable> firstFailure = new AtomicReference<>();
+    private final AtomicReference<Throwable> firstFailureRef = new AtomicReference<>();
     private final CountDownLatch done = new CountDownLatch(1);
     private volatile boolean sealed;
 
@@ -299,13 +299,13 @@ public class HiveDriverPool implements AutoCloseable {
      * later {@code setException} into a no-op and the error into a CancellationException.
      */
     private void recordFailure(Throwable t) {
-      firstFailure.compareAndSet(null, t);
+      firstFailureRef.compareAndSet(null, t);
       aborted.set(true);
       done.countDown();
     }
 
     private Throwable firstFailure() {
-      return firstFailure.get();
+      return firstFailureRef.get();
     }
 
     private void taskSettled() {

--- a/hudi-sync/hudi-hive-sync/src/test/java/org/apache/hudi/hive/util/TestHiveDriverPool.java
+++ b/hudi-sync/hudi-hive-sync/src/test/java/org/apache/hudi/hive/util/TestHiveDriverPool.java
@@ -40,6 +40,7 @@ import java.util.concurrent.atomic.AtomicInteger;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.anyString;
@@ -138,7 +139,8 @@ class TestHiveDriverPool {
       HiveDriverPool.Dispatch futures = pool.dispatchAll(Arrays.asList("OK", "FAIL", "OK"));
       HoodieHiveSyncException ex = assertThrows(HoodieHiveSyncException.class,
           () -> pool.awaitAll(futures));
-      assertTrue(ex.getCause() != null && ex.getCause().getMessage().contains("boom"));
+      assertNotNull(ex.getCause());
+      assertTrue(ex.getCause().getMessage().contains("boom"));
     }
   }
 
@@ -254,7 +256,8 @@ class TestHiveDriverPool {
       HoodieHiveSyncException ex = assertThrows(HoodieHiveSyncException.class,
           () -> pool.awaitAll(dispatch));
 
-      assertTrue(ex.getCause() != null && ex.getCause().getMessage().contains("boom"));
+      assertNotNull(ex.getCause());
+      assertTrue(ex.getCause().getMessage().contains("boom"));
       assertEquals(Collections.singletonList("FAIL"), executed,
           "Statements queued behind the failure must never reach the Driver");
     }
@@ -307,7 +310,8 @@ class TestHiveDriverPool {
       HoodieHiveSyncException ex = assertThrows(HoodieHiveSyncException.class,
           () -> pool.awaitAll(dispatch));
 
-      assertTrue(ex.getCause() != null && ex.getCause().getMessage().contains("boom"));
+      assertNotNull(ex.getCause());
+      assertTrue(ex.getCause().getMessage().contains("boom"));
       assertFalse(executed.contains("AFTER_FAIL"),
           "Statement queued behind a failure on the same worker must not be applied, "
               + "even while an earlier future on another worker is still running");
@@ -353,7 +357,8 @@ class TestHiveDriverPool {
 
       HoodieHiveSyncException ex = assertThrows(HoodieHiveSyncException.class,
           () -> pool.awaitAll(dispatch));
-      assertTrue(ex.getCause() != null && ex.getCause().getMessage().contains("boom"));
+      assertNotNull(ex.getCause());
+      assertTrue(ex.getCause().getMessage().contains("boom"));
     }
   }
 }

